lib/ansible/modules/cloud/alicloud/alicloud_instance.py

```diff
@@ -0,0 +1,584 @@
+      default: 0
+    host_name:
+      description:
+        - Instance host name.
+    password:
+      description:
+        - The password to login instance. After rebooting instances, the modified password would be take effect.
+    system_disk_category:
+      description:
+        - Category of the system disk.
+      default: 'cloud_efficiency'
+      choices: ['cloud_efficiency', 'cloud_ssd']
+    system_disk_size:
+      description:
+        - Size of the system disk, in GB. The valid values are 40~500.
+      default: 40
+    system_disk_name:
+      description:
+        - Name of the system disk.
+    system_disk_description:
+      description:
+        - Description of the system disk.
+    count:
+      description:
+        - The number of the new instance. An integer value which indicates how many instances that match I(count_tag)
+          should be running. Instances are either created or terminated based on this value.
+      default: 1
+    count_tag:
+      description:
+      - I(count) determines how many instances based on a specific tag criteria should be present.
+        This can be expressed in multiple ways and is shown in the EXAMPLES section.
+        The specified count_tag must already exist or be passed in as the I(instance_tags) option.
+        If it is not specified, it will be replaced by I(instance_name).
+    allocate_public_ip:
+      description:
+        - Whether allocate a public ip for the new instance.
+      default: False
+      aliases: [ 'assign_public_ip' ]
+      type: bool
+    instance_charge_type:
+      description:
+        - The charge type of the instance.
+      choices: ['PrePaid', 'PostPaid']
+      default: 'PostPaid'
+    period:
+      description:
+        - The charge duration of the instance, in month. Required when C(instance_charge_type=PrePaid).
+        - The valid value are [1-9, 12, 24, 36].
+      default: 1
+    auto_renew:
+      description:
+        - Whether automate renew the charge of the instance.
+      type: bool
+      default: False
+    auto_renew_period:
+      description:
+        - The duration of the automatic renew the charge of the instance. Required when C(auto_renew=True).
+      choices: [1, 2, 3, 6, 12]
+    instance_ids:
+      description:
+        - A list of instance ids. It is required when need to operate existing instances.
+          If it is specified, I(count) will lose efficacy.
+    force:
+      description:
+        - Whether the current operation needs to be execute forcibly.
+      default: False
+      type: bool
+    instance_tags:
+      description:
+        - A hash/dictionaries of instance tags, to add to the new instance or for starting/stopping instance by tag. C({"key":"value"})
+      aliases: ["tags"]
+    sg_action:
+      description:
+        - The action of operating security group.
+      choices: ['join', 'leave']
```

Comment:
Adding/removing security groups is usually implemented like this:

```
groups:
- {identifier of security group}
- {identifier of second security group}
purge_groups: true/false
```
That way if a user specifies a set of groups, the instance will be added if they are not already on it. If they want to leave groups, they specify the ones that must remail and then set `purge_groups: true`. For an example, see the `ec2_group` module.